client/src/utils/contexts/context.js

```diff
@@ -0,0 +1,38 @@
+import React, { useState } from "react";
+import PropTypes from "prop-types";
+
+const UserContext = React.createContext();
+
+export const useUserState = () => {
+  const context = React.useContext(UserContext);
+  if (context === undefined) {
+    throw new Error("useUserState must be used within a UserProvider");
+  }
+  return context;
```

Comment:
I think you can dismiss the React. :
```suggestion
const UserContext = createContext();

export const useUserState = () => {
  const context = useContext(UserContext);
  if (context === undefined) {
    throw new Error("useUserState must be used within a UserProvider");
  }
  return context;
```